[ADDED] Allowed connection type "in process"

kozlovic opened the pull request:
With the JWT library update, there is now a `jwt.ConnectionTypeInProcess` connection type that can be added to the list of allowed connection types when defining an user. If the list is not empty and does not have this type, an "in process" connection will not be allowed.

Conversely, it is now possible to limit a user to be used only for in process connections (or several types based on the content of the list).

Possible solution for #5387

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

kozlovic commented:
@derekcollison This would be a possible solution to issue #5387. If you think this is reasonable, I will make the JWT PR, have it merged, and then update this PR with the new JWT lib dep and open this for formal review.

derekcollison commented:
Will take a look.

derekcollison reviewed (commented):
LGTM - Let's proceed.

kozlovic commented:
@derekcollison I have updated the PR with the go.mod change to point to newly released JWT library that adds the "in process" connection type and updated the test to cover JWT test. It is ready for review now.

derekcollison reviewed (approved):
LGTM
